source/codegen/templates/proto_helpers.mako

```diff
@@ -51,45 +19,60 @@ enum ${service_class_prefix}Attributes {
   ${attribute_value_prefix}_${attribute_name} = ${attribute};
 % endfor
 }
+</%def>
 
+## Define enums in the proto for each metadata enum referenced in a proto message.
+<%def name="define_enums(used_enums)">\
+<%
+  enums = data["enums"]
+%>\
 % for enum_name in (e for e in enums if e in used_enums):
 <%
   enum_value_prefix = common_helpers.pascal_to_snake(enum_name).upper()
   enum = enums[enum_name]
-  allow_alias = proto_helpers.determine_allow_alias(enum)
+  allow_alias = proto_helpers.should_allow_alias(enum)
+  nonint_index = 1
 %>\
 enum ${enum_name} {
 % if allow_alias == True:
   option allow_alias = true;
 % endif
   ${enum_value_prefix}_UNSPECIFIED = 0;
-<%
-nonint_index = 1
-%>\
 % for value in enum["values"]:
-% if enum.get("generate-mappings", false):
+% if enum.get("generate-mappings", False):
   ${enum_value_prefix}_${value["name"]} = ${nonint_index};
 <%
-  nonint_index = nonint_index+1
+  nonint_index = nonint_index + 1
 %>\
 % else:
   ${enum_value_prefix}_${value["name"]} = ${value["value"]};
-%endif
+% endif
 % endfor
 }
 
 % endfor
-%if has_custom_template:
+</%def>
+
+## If there is a custom proto mako template in the metadata, insert it.
+<%def name="insert_custom_template_if_found()">\
+<%
+  config = data["config"]
+  lookup = data["lookup"]
+  has_custom_template = False
+  if len(config["custom_types"]) > 0:
```

Comment:
```suggestion
  if config["custom_types"]:
```
In python, objects with a __len__() method have implicit boolean conversions and I think you should be able to use that here. See https://docs.python.org/3/library/stdtypes.html#truth-value-testing

Reply:
Oh neat. Committed the suggested change.